File: BlogLite-Server/src/main/java/com/wsmrxd/bloglite/service/UserService.java
package com.wsmrxd.bloglite.service;

import com.wsmrxd.bloglite.entity.User;
import com.wsmrxd.bloglite.vo.LoginSuccessInfo;

public interface UserService {
    User getUser(int id);

    User getUser(String email);

    int queryUserID(String email);

    boolean changePassword(String email, String newPassword);

    boolean changeUsername(String email, String newUsername);

    LoginSuccessInfo doLogin(String email, String password);
}

File: BlogLite-Server/src/main/java/com/wsmrxd/bloglite/service/UserService.kt
package com.wsmrxd.bloglite.service

import com.wsmrxd.bloglite.entity.User
import com.wsmrxd.bloglite.vo.LoginSuccessInfo

interface UserService {
    fun getUser(email: String): User?

    fun doLogin(email: String, password: String): LoginSuccessInfo
}
